### PR TITLE
Energy setup wizard missing localization entries

### PR DESCRIPTION
**Pull request by yosilevy**
## Proposed change

The Step 1 of 5 text in the energy setup wizard was hardcoded. Added to translations.

- [X ] Bugfix (non-breaking change which fixes an issue)

- [X ] The code change is tested and works locally.
- [ X] There is no commented out code in this PR.

### CODE REVIEW

**Comment by spacegaier** [comment]
Can you please keep the PR template as-is. And there may not be any spaces in the square brackets, otherwise the checkboxes will not render here.

File: src/translations/en.json [hunk 1]
```diff
@@ -4096,7 +4096,9 @@
         "setup": {
           "next": "Next",
           "back": "Back",
-          "done": "Show me my energy dashboard!"
+          "done": "Show me my energy dashboard!",
+          "step": "Step",
+          "of": "of"
```
Comment (spacegaier) [inline src/translations/en.json:4101]:
```suggestion
          "step_of": "of"
```
I would slightly prefer that so it more closely ties to the "step" key.

Comment (yosilevy) [reply src/translations/en.json]: Done

**Review by zsarnett — COMMENTED** [review]
Can we instead use variables in the localization? 

```js
this.hass.localize("ui.panel.energy.setup.step", "step", 1, "step_of", 5)
```

```json
"step_of": "Step {step} of {step_of}"
```

**Comment by spacegaier** [comment]
> Can we instead use variables in the localization?
> 
> ```js
> this.hass.localize("ui.panel.energy.setup.step", "step", 1, "step_of", 5)
> ```
> 
> ```json
> "step_of": "Step {step} of {step_of}"
> ```

Yes, that is even better (why didn't I think that of 😕?). Removes some ambiguities for the community translations.

**Comment by yosilevy** [comment]
Refactored to 
```
        ${this.hass.localize("ui.panel.energy.setup.step", {
          step: this._step + 1,
          steps: 5,
        })}
```

**Review by spacegaier — APPROVED** [review]

**Review by zsarnett — APPROVED** [review]

**Pull request merged**